File: solutions/javascript/0700-search-in-a-binary-search-tree.js
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {
    if (root === null) {
        return null;
    }
    if (root.val === val) {
        return root;
    }
    if (root.val < val) {
        return searchBST(root.right, val);
    }
    return searchBST(root.left, val);
};

File: solutions/cpp/0700-search-in-a-binary-search-tree.cpp
class Solution {
public:
    TreeNode* searchBST(TreeNode* root, int val) {
        if (root == NULL) {
            return NULL;
        }
        if (root->val == val) {
            return root;
        }
        if (root->val < val) {
            return searchBST(root->right, val);
        }
        return searchBST(root->left, val);
    }
};
